bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/osgi/ResourceBundleClassLoader.java

```diff
@@ -36,10 +36,15 @@
  * mappings.
  *
  * @author Michael Grammling - Initial Contribution
+ * @author Martin Herbst - ISO-8859-1 support added
  */
 public class ResourceBundleClassLoader extends ClassLoader {
 
-    private static final String[] SUPPORTED_CHARSETS = { "UTF-8" };
+    /**
+     * ISO-8859-1 must be the last array element because all files could be decoded to ISO-8859-1 and
+     * therefore always character set ISO-8859-1 would be assumed.
```

Comment:
Hm, I do not understand this. Does this mean that decoder.decode() will never throw a CharacterCodingException, since every byte is a valid character in ISO-8859-1? And UTF-8 decoding will fail on any non-ASCII character, because these bytes are not valid in an UTF-8 byte array?

Would it be possible to add some unit test in org.eclipse.smarthome.core.test that actually tests this charset detection mechanism? I would feel more comfortable with that :-)